jsapp/js/account/organization/OrganizationSettingsRoute.tsx

```diff
@@ -1,7 +1,156 @@
+ * (with some logic in regards to their visibility). If user has necessary role,
+ * they can edit available fields.
+ */
 export default function OrganizationSettingsRoute() {
+  const orgQuery = useOrganizationQuery();
+  const [subscriptions] = useState(() => subscriptionStore);
+  const [state, setState] = useState<State>({name: ''});
+  const [isStripeEnabled, setIsStripeEnabled] = useState(false);
+
+  useEffect(() => {
+    if (orgQuery.data) {
+      setState({name: orgQuery.data.name});
+    }
+  }, [orgQuery.data]);
+
+  useWhenStripeIsEnabled(() => {
+    setIsStripeEnabled(true);
+  }, []);
+
+  const isUserAdminOrOwner = (
+    orgQuery.data?.request_user_role &&
+    [OrganizationUserRole.admin, OrganizationUserRole.owner]
+      .includes(orgQuery.data?.request_user_role)
+  );
+
+  const isPendingOrgPatch = orgQuery.data && orgQuery.isPending;
+
+  function handleSave() {
+    // TODO: call the API endpoint
+    // to be done while doing: https://www.notion.so/kobotoolbox/Add-react-query-mutation-hook-for-org-changes-1307e515f6548010b5d3c087b634f01a
+    console.log('save');
+  }
+
+  function handleChangeName(name: string) {
+    setState((prevState) => {return {...prevState, name};});
+  }
+
+  function handleChangeWebsite(website: string) {
+    setState((prevState) => {return {...prevState, website};});
+  }
+
+  function isNameValueValid(currentName: string) {
+    return !currentName;
+  }
+
+  function isWebsiteValueValid(currentWebsite: string) {
+    return !currentWebsite;
+  }
+
+  function getTypeLabel(typeName: string) {
+    // TODO: see if this would be an actual source of the organization type label
+    // to be done while doing: https://www.notion.so/kobotoolbox/Add-react-query-mutation-hook-for-org-changes-1307e515f6548010b5d3c087b634f01a
+    const foundLabel = ORGANIZATION_TYPE_SELECT_OPTIONS.find((item) => item.value === typeName)?.label;
+    return foundLabel || typeName;
+  }
+
+  const mmoLabel = getSimpleMMOLabel(
+    envStore.data,
+    subscriptionStore.activeSubscriptions[0],
+    false,
+    true
+  );
+  const mmoLabelLowercase = mmoLabel.toLowerCase();
+
+  if (!orgQuery.data) {
+    return <LoadingSpinner />;
+  }
+
   return (
-    <div>Organization settings view to be implemented</div>
+    <div className={styles.orgSettingsRoot}>
+      <header className={styles.orgSettingsHeader}>
+        <h2 className={styles.orgSettingsHeaderText}>
+          {t('##team/org## details').replace('##team/org##', mmoLabel)}
+        </h2>
+      </header>
+
+      <section className={styles.fieldsRow}>
+        <OrganizationSettingsField
+          label={t('##team/org## name').replace('##team/org##', mmoLabel)}
+          onChange={handleChangeName}
+          value={state.name}
+          validateValue={isNameValueValid}
+          isDisabled={!isUserAdminOrOwner || isPendingOrgPatch}
+        />
+
+        {isStripeEnabled && state.website && (
```

Comment:
For me, who doesn't have the `website` filled in yet, the field is not displayed for editing. Is this on purpose? 
Same thing for the `type`